Cython/Compiler/Symtab.py

```diff
@@ -1933,7 +1940,8 @@ def declare_var(self, name, type, pos,
 
     def declare_cfunction(self, name, type, pos,
                           cname=None, visibility='private', api=0, in_pxd=0,
-                          defining=0, modifiers=(), overridable=False):  # currently no utility code ...
+                          defining=0, modifiers=(), overridable=False,
+                          is_property=False):  # currently no utility code ...
```

Comment:
Same thing here, I'd rather handle `is_property` the same as the `utility_code` argument for scopes that do not support it, i.e. make it a hard error if we end up passing it, by simply not allowing it.

Reply:
The call chain is `Nodes.CFuncDefNode.analyse_declarations` finds the property decorator and later calls `env.declare_cfunction` where `env` is any one of `Scope`, `CClassScope` or .`CppClassScope`. If I remove the kwarg from `Symtab.Scope`, how do I adjust the call in `analyse_declarations`?

Also I see that I never handle the kwarg in `Symtab.CppClassScope.declare_cfunction`. Do I need to add a test for a `CppClass` that uses a `@property to define a getter? I didn't really grok that `CppClass` is different from `CClass`

Reply:
After staring at the examples of `cdef cppclass` I think use of a `@property` decorator to replace a `cdef attribute` getter with a c function is not needed, so the code path in `Symtab.CppClassScope.declare_cfunction` should never have `is_property` be True. Do you agree, and should I assert `is_property == False` at the top of this function?

Reply:
ok, removed the kwarg, at the cost of moving the entry from one property list to another after its creation

Reply:
just now, 261ad4d